tests/test_database.py

```
@@ -93,6 +104,34 @@ def test_run_startup_migrations_with_existing_tables_no_alembic_version(
-    mock_stamp.assert_called_once_with(ANY, "head")
+    # The max_concurrent_migrations column tells us this schema matches the
+    # pre-head revision immediately before RelocationTask.
+    mock_stamp.assert_called_once_with(ANY, "6b398cde9d3e")
+    mock_upgrade.assert_called_once_with(ANY, "head")
+
+
+@patch("alembic.command.upgrade")
+@patch("alembic.command.stamp")
+@patch("app.database.engine")
+def test_run_startup_migrations_stamps_max_concurrent_schema_to_matching_revision(
+    mock_engine, mock_stamp, mock_upgrade, db_session, monkeypatch
```

Ruff (0.15.9)

[warning] Unused function argument: `db_session`

(ARG001)